pkg/restic/common.go

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright 2018 the Heptio Ark contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package restic
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	arkv1client "github.com/heptio/ark/pkg/apis/ark/v1"
+)
+
+const (
+	podAnnotationPrefix         = "snapshot.ark.heptio.com/"
+	volumesToBackupAnnotation   = "backup.ark.heptio.com/backup-volumes"
+	snapshotsInBackupAnnotation = "backup.ark.heptio.com/restic-snapshots"
+)
+
+// TODO audit functions, make private (?)
+
+func PodHasSnapshotAnnotation(obj metav1.Object) bool {
+	for key := range obj.GetAnnotations() {
+		if strings.HasPrefix(key, podAnnotationPrefix) {
+			return true
+		}
+	}
+
+	return false
+}
+
+func GetPodSnapshotAnnotations(obj metav1.Object) map[string]string {
+	var res map[string]string
+
+	for k, v := range obj.GetAnnotations() {
+		if strings.HasPrefix(k, podAnnotationPrefix) {
+			if res == nil {
+				res = make(map[string]string)
+			}
+
+			res[k[len(podAnnotationPrefix):]] = v
+		}
+	}
+
+	return res
+}
+
+func SetPodSnapshotAnnotation(obj metav1.Object, volumeName, snapshotID string) {
+	annotations := obj.GetAnnotations()
+
+	if annotations == nil {
+		annotations = make(map[string]string)
+	}
+
+	annotations[podAnnotationPrefix+volumeName] = snapshotID
+
+	obj.SetAnnotations(annotations)
+}
+
+func GetVolumesToBackup(obj metav1.Object) []string {
+	annotations := obj.GetAnnotations()
+	if annotations == nil {
+		return nil
+	}
+
+	backupsValue := annotations[volumesToBackupAnnotation]
+	if backupsValue == "" {
+		return nil
+	}
+
+	var backups []string
+	// check for json array
```

Comment:
Would it make more sense to do a comma-separated string instead of a json-encoded array?